dlc-test/src/test/scala/org/bitcoins/dlc/NumericDLCTest.scala

```diff
@@ -0,0 +1,132 @@
+package org.bitcoins.dlc
+
+import org.bitcoins.core.protocol.tlv.{
+  OracleParamsV0TLV,
+  UnsignedNumericOutcome
+}
+import org.bitcoins.core.util.NumberUtil
+import org.bitcoins.testkitcore.dlc.DLCTest
+import org.bitcoins.testkitcore.util.BitcoinSJvmTest
+
+import scala.util.Random
+
+class NumericDLCTest extends BitcoinSJvmTest with DLCTest {
+  behavior of "Numeric DLC"
+
+  val numericOracleSchemesToTest: Vector[(Int, Int)] =
+    Vector((1, 1), (2, 2), (2, 3))
+  val numDigitsToTest: Vector[Int] = Vector(4, 5, 10)
+
+  it should "be able to construct and verify with ScriptInterpreter every tx in a DLC for the normal numeric case" in {
+    runTestsForParam(numDigitsToTest) { numDigits =>
+      runTestsForParam(numericOracleSchemesToTest) {
+        case (threshold, numOracles) =>
+          val contractParams =
+            NumericContractParams(numDigits, threshold, numOracles)
+          val nums = tenRandomNums(numDigits)
+
+          executeForCases(nums, contractParams)
+      }
+    }
+  }
+
+  it should "be able to construct and verify with ScriptInterpreter every tx in a DLC for the large numeric case" in {
+    val numDigits = 17
+    val contractParams =
+      NumericContractParams(numDigits, oracleThreshold = 1, numOracles = 1)
+    val nums = tenRandomNums(numDigits)
+
+    executeForCases(nums, contractParams)
+  }
```

Comment:
This seems to be timing out on CI in the non-secp-binary setting so perhaps it is worth it to ratchet down `numDigits` when secp is not enabled?